web3/middleware/gas_price_strategy.py

```diff
@@ -27,11 +32,44 @@ def gas_price_strategy_middleware(
     def middleware(method: RPCEndpoint, params: Any) -> RPCResponse:
         if method == 'eth_sendTransaction':
             transaction = params[0]
-            if 'gasPrice' not in transaction:
-                generated_gas_price = web3.eth.generate_gas_price(transaction)
+
+            # legacy and 1559 tx variables used:
+            if "gasPrice" in transaction and (
+                "maxFeePerGas" in transaction or "maxPriorityFeePerGas" in transaction
+            ):
+                raise TransactionTypeMismatch()
+            # 1559 - canonical tx:
+            elif 'maxFeePerGas' in transaction and 'maxPriorityFeePerGas' in transaction:
+                if int(transaction["maxFeePerGas"], 16) < int(
+                    transaction["maxPriorityFeePerGas"], 16
+                ):
+                    raise InvalidTransaction("maxFeePerGas must be >= maxPriorityFeePerGas")
+            # 1559 - no feecap:
+            elif 'maxFeePerGas' not in transaction and 'maxPriorityFeePerGas' in transaction:
+                block = web3.eth.get_block('latest')
+                base_fee = block['baseFeePerGas']
+                tip = Wei(int(transaction['maxPriorityFeePerGas'], 16))
+                if base_fee < tip:
+                    # TODO: throw or just set feecap to max prio?
+                    base_fee = tip
+                    #  raise InvalidTransaction('maxFeePerGas must be >= maxPriorityFeePerGas')
+                transaction = assoc(transaction, 'maxFeePerGas', hex(base_fee * 2))
+                return make_request(method, [transaction])
+            # 1559 - no tip:
+            elif 'maxFeePerGas' in transaction and 'maxPriorityFeePerGas' not in transaction:
+                raise InvalidTransaction(
+                    "maxPriorityFeePerGas must be defined in a 1559 transaction."
+                )
+            # legacy tx:
+            elif 'gasPrice' not in transaction:
```

Comment:
Should this be put back before merging? I think this changes the default behavior if I'm reading it right?

Reply:
Yeah, I think that's right. This is a noop anyway, so will just nix this condition now.